OpenRA.Mods.Common/Traits/Repairable.cs

```diff
@@ -108,5 +110,46 @@ public void ResolveOrder(Actor self, Order order)
 					}));
 			}
 		}
+
+		public Actor FindRepairBuilding(Actor self)
+		{
+			var repairBuilding = self.World.ActorsWithTrait<RepairsUnits>()
+			                          .Where(a => !a.Actor.IsDead && a.Actor.IsInWorld
+			                                      && a.Actor.Owner == self.Owner &&
+			                                      info.RepairBuildings.Contains(a.Actor.Info.Name))
+			                          .OrderBy(p => (self.Location - p.Actor.Location).LengthSquared);
+
+			return repairBuilding.FirstOrDefault().Actor;
+		}
+
+		void TryCallTransport(Actor self, Target target, IMove movement)
+		{
+			var transport = self.TraitOrDefault<ICallForTransport>();
+			if (transport == null)
+				return;
+
+			var targetCell = self.World.Map.CellContaining(target.CenterPosition);
+			if ((self.Location - targetCell).LengthSquared < transport.MinimumDistance * transport.MinimumDistance)
+				return;
+
+			// TODO: This is hacky, but almost every single component affected
+			// will need to be rewritten anyway, so this is OK for now.
+			var nextActivity = new MoveAdjacentTo(self, target);
+			nextActivity.Queue(movement.MoveTo(targetCell, target.Actor));
```

Comment:
It can drop them directly on it, but if it's occupied it will put them somewhere on the side. If this activity is missing, they will start repairing somewhere around the repair pad (just another issue caused by the less-than-nice repair code).